checkov/cloudformation/graph_builder/graph_to_definitions.py

```diff
@@ -14,7 +14,7 @@ def convert_graph_vertices_to_definitions(
     for vertex in vertices:
         block_path = vertex.path
         block_type = CloudformationTemplateSections.RESOURCES.value if vertex.block_type == 'resource' else vertex.block_type
-        block_name = vertex.name.split('.')[1]  # vertex.name is "type.name" so type.name -> [type, name]
+        block_name = vertex.name.split('.')[-1]  # vertex.name is "type.name" so type.name -> [type, name]
```

Comment:
What’s the purpose of this change?

Reply:
It handles a special case where a resource's `Type: Boto3::CodeCommit.put_file`
When we create the block name it will be: `Boto3::CodeCommit.put_file.<name>`
If we use `[1]` we will get `block_name='put_file'`, if we use `[-1]` instead it will use the last index in the array